games-action/polymc/polymc-1.2.1.ebuild

```diff
@@ -0,0 +1,111 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake java-pkg-2 optfeature xdg
+
+HOMEPAGE="https://polymc.org/ https://github.com/PolyMC/PolyMC"
+DESCRIPTION="A custom, open source Minecraft launcher"
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+
+	EGIT_REPO_URI="
+		https://github.com/PolyMC/PolyMC
+		https://github.com/PolyMC/libnbtplusplus
+	"
+
+	EGIT_SUBMODULES=( 'depends/libnbtplusplus' )
+else
+	MY_PN="PolyMC"
+
+	# Let's use the vendored tarball to avoid dealing with the submodules directly
+	SRC_URI="
+		https://github.com/PolyMC/PolyMC/releases/download/${PV}/${MY_PN}-${PV}.tar.gz -> ${P}.tar.gz
+	"
+
+	# The PolyMC's files are unpacked to ${WORKDIR}/PolyMC-${PV}
+	S="${WORKDIR}/${MY_PN}-${PV}"
+
+	KEYWORDS="~amd64"
+fi
+
+# Apache-2.0 for MultiMC (PolyMC is forked from it)
+# GPL-3 for PolyMC
+# LGPL-3 for libnbtplusplus
+# LGPL-2.1 with linking exception for Quazip
+# See the rest of PolyMC's libraries at https://github.com/PolyMC/PolyMC/tree/develop/libraries
+LICENSE="Apache-2.0 Boost-1.0 BSD BSD-2 GPL-2+ GPL-3 LGPL-3 LGPL-2.1-with-linking-exception OFL-1.1 MIT"
+
+SLOT="0"
+
+IUSE="debug lto"
+REQUIRED_USE="
+	lto? ( !debug )
+"
+
+MIN_QT="5.6.0"
+
+QT_DEPS="
+	>=dev-qt/qtconcurrent-${MIN_QT}:5
+	>=dev-qt/qtcore-${MIN_QT}:5
+	>=dev-qt/qtgui-${MIN_QT}:5
+	>=dev-qt/qtnetwork-${MIN_QT}:5
+	>=dev-qt/qttest-${MIN_QT}:5
+	>=dev-qt/qtwidgets-${MIN_QT}:5
+	>=dev-qt/qtxml-${MIN_QT}:5
+"
+
+DEPEND="
+	${QT_DEPS}
+	>=dev-libs/quazip-1.3
+	media-libs/libglvnd
+	sys-libs/zlib
+	>=virtual/jdk-1.8.0:*
+"
+
+# At run-time we don't depend on JDK, only JRE
+# And we need more than just the GL headers
+RDEPEND="
+	${QT_DEPS}
+	sys-libs/zlib
+	>=virtual/jre-1.8.0:*
+	virtual/opengl
+"
+
+src_prepare() {
+	cmake_src_prepare
+}
+
+src_configure(){
+	local mycmakeargs=(
+		-DCMAKE_INSTALL_PREFIX="/usr"
+		# Resulting binary is named polymc
+		-DLauncher_APP_BINARY_NAME="${PN}"
+
+		-DENABLE_LTO=$(usex lto)
+
+		# Needed by libraries/iconfix
+		-DBUILD_SHARED_LIBS=0
```

Comment:
For some reason, the iconfix .so file is not recognized by PolyMC without this option, which is weird given that it's the only one with a problem. I don't yet know if it's an upstream issue, since they use this option on the CI, but my local tests required it. I'll see if I can fix this, or maybe only build as static lib this one library!

Reply:
Sounds like an upstream issue, we don't have this problem with version 1.1.1. Something must have changed, I'll see if I can find out what it is.

Reply:
We didn't have this problem in 1.1.1 because, when building with bundled Quazip, this option was set automatically!

Reply:
Actually, now that I see, most libraries in https://github.com/PolyMC/PolyMC/tree/develop/libraries are linked statically by default, except javacheck and optional-bare it seems. Maybe it is an upstream problem then, it just never showed up because support for system Quazip was added just a few days ago

Reply:
> We didn't have this problem in 1.1.1 because, when building with bundled Quazip, this option was set automatically!

Aaah now it makes sense. So the upstream build system should probably have some logic to handle the `-DBUILD_SHARED_LIBS=1` case that is now possible (i.e. actually install the `iconfix.so` shared library to the file system if shared libs are enabled, or your current patch to just always use the iconfix library statically, either will work).

>  I don't yet know if it's an upstream issue, since they use this option on the CI, but my local tests required it. I'll see if I can fix this, or maybe only build as static lib this one library! 

This won't show up in CI because there is no issue compiling, linking or testing with the library because it is built correctly and it is present during testing. The only problem is that the library is not installed to the file system so when we actually want to run polymc we cannot find it.

Reply:
> This won't show up in CI because there is no issue compiling, linking or testing with the library because it is built correctly and it is present during testing. The only problem is that the library is not installed to the file system so when we actually want to run polymc we cannot find it.

Yes, what I meant to say is that people using the builds from the CI wouldn't notice it because the CI uses bundled quazip for building, so it automatically sets the static libs flag and builds iconfix statically as well :smile: 

> Aaah now it makes sense. So the upstream build system should probably have some logic to handle the -DBUILD_SHARED_LIBS=1 case that is now possible (i.e. actually install the iconfix.so shared library to the file system if shared libs are enabled, or your current patch to just always use the iconfix library statically, either will work).

Yes, I'll submit the patch for always building it as an static library, since most other libraries already do this, and adding the logic for using them as shared libraries will require more work, and I don't know if they even want to ([they _did_ a PR to build libraries statically a while ago](https://github.com/PolyMC/PolyMC/pull/143) ).